ow_plexil/src/plans/ArmFindSurface.plp

```diff
@@ -0,0 +1,50 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// Perform an ArmFindSurface as specified.
+
+#include "plan-interface.h"
+
+ArmFindSurface:
+{
+  In Integer Frame;
+  In Boolean Relative;
+  In Real PosX;
+  In Real PosY;
+  In Real PosZ;
+  In Real NormX;
+  In Real NormY;
+  In Real NormZ;
+  In Real Distance;
+  In Real Overdrive;
+  In Real ForceThreshold;
+  In Real TorqueThreshold;
+
+  Boolean FaultDetected = false;
+
+  PostCondition !Lookup(ArmFault);
+
+  if Lookup(ArmFault)
+  {
+    log_error ("Command arm_find_surface not sent ",
+	       "to lander due to active arm fault(s).");
+    FaultDetected = true;
+  }
+
+  SendArmFindSurface:
+  {
+    Start !Lookup(ArmFault);
+
+    if FaultDetected
+    {
+      log_info ("Arm fault(s) resolved, ",
+		"sending arm_find_surface command to lander...");
+    }
```

Comment:
This pattern was written by an intern a couple summers ago, and is found in many library plans.  The pattern reads: If there's an active fault at the time the command is requested, wait indefinitely for the fault to clear before sending the command (and issue messages that there was a fault, and that it was cleared).  Otherwise, just send the command immediately.

This isn't the logic that some users will want, but they can use custom Plexil instead of these library nodes.

Reply:
Ah, I see that now. Thanks. It just takes some time to remember how logic flows in PLEXIL.